moduleframework/helpers/openshift_helper.py

```diff
@@ -0,0 +1,268 @@
+            pod_state = pod_state.stdout.split('\n')
+            for pod in pod_state:
+                if pod.startswith(self.app_name):
+                    if "Running" in pod and "deploy" not in pod:
+                        pod_initiated = True
+                        break
+            time.sleep(1)
+            if pod_initiated:
+                break
+        return pod_initiated
+
+    def setUp(self):
+        """
+        It is called by child class and it is same methof as Avocado/Unittest has. It prepares environment
+        for docker testing
+        * start docker if not
+        * pull docker image
+        * setup environment from config
+        * run and store identification
+
+        :return: None
+        """
+        self.icontainer = self.get_url()
+
+    def _openshift_login(self, oc_ip="127.0.0.1", oc_user='developer', oc_passwd='developer', env=False):
+        """
+        It logins to an OpenShift environment on specific IP and under user and his password.
+        :param oc_ip: an IP where is an OpenShift environment running
+        :param oc_user: an username under which we can login to OpenShift environment
+        :param oc_passwd: a password for specific username
+        :param env:
+        :return:
+        """
+        if env:
+            if 'OPENSHIFT_IP' in os.environ:
+                oc_ip = os.environ.get('OPENSHIFT_IP')
+            if 'OPENSHIFT_USER' in os.environ:
+                oc_user = os.environ.get('OPENSHIFT_USER')
+            if 'OPENSHIFT_PASSWORD' in os.environ:
+                oc_passwd = os.environ.get('OPENSHIFT_PASSWORD')
+
+        oc_output = self.runHost("oc login %s:8443 --username=%s --password=%s" % (oc_ip,
+                                                                                   oc_user,
+                                                                                   oc_passwd),
+                                 verbose=common.is_not_silent())
+        common.print_debug(oc_output.stdout)
+        return oc_output.exit_status
+
+    def tearDown(self):
+        """
+        Cleanup environment and call also cleanup from config
+
+        :return: None
+        """
+        super(OpenShiftHelper, self).tearDown()
+        if common.get_if_do_cleanup():
+            # TODO will be implemented later on. I have to find a usecase what to remove
+            pass
+
+    def _get_ip_instance(self):
+        """
+        It gets and IP address of an application from of OpenShift POD.
+        :return: True: getting IP address was successful
+                 False: getting IP address was not successful
+        """
+        oc_get_service = self.runHost("oc get service -o json")
+        service = self._convert_string_to_json(oc_get_service.stdout)
+        try:
+            for svc in service["items"]:
+                if "clusterIP" in svc.get("spec"):
+                    common.trans_dict['GUESTIPADDR'] = svc.get("spec").get("clusterIP")
```

Comment:
maybe this is proper place where to modify also `self.ipaddr` variable and then you can remove some special handling of ipadd in common.CommonFunciton for OS

and maybe question, why there is __for__ cycle. means that there can be more ipadresses and and the last one is the best one?
or you meant code like:
```
for svc un service.get("items", []):
    ipaddr = svc.get("spec",{}).get("clusterIP")
    if ipaddr:
          common.trans_dict['GUESTIPADDR'] = ipaddr
          self.ipaddr = ipaddr
          return True
return False
```
?